### PR TITLE
Improve generate template command

### PR DESCRIPTION
**Pull request by williamjamir**
Adds three new parameter for Generate plugin template command

 - extra_includes: Added custom includes on the foo.cpp
 - extra_body_lines: Add extra lines to on foo.cpp, used for default implementations of hooks
 - exclude_hooks: List of hooks, that will not be inserted on the foo.cpp

### CODE REVIEW

File: hookman/hookman_generator.py [hunk 1]
```diff
@@ -167,14 +179,35 @@ def generate_plugin_template(self,
...
         Path(plugin_folder / 'compile.py').write_text(self._compile_shared_lib_python_script_content(plugin_id))
         Path(plugin_folder / 'CMakeLists.txt').write_text(self._plugin_cmake_file_content(plugin_id))
         Path(assets_folder / 'plugin.yaml').write_text(self._plugin_config_file_content(caption, plugin_id, author_email, author_name))
         Path(assets_folder / 'README.md').write_text(self._readme_content(caption, author_email, author_name))
         Path(source_folder / 'hook_specs.h').write_text(self._hook_specs_header_content(plugin_id))
-        Path(source_folder / 'plugin.c').write_text(self._plugin_source_content())
+        Path(source_folder / f'{plugin_id}.cpp').write_text(self._plugin_source_content(extra_includes, extra_body_lines, exclude_hooks))
         Path(source_folder / 'CMakeLists.txt').write_text(self._plugin_src_cmake_file_content(plugin_id))
 
+    def _parameter_validation(self, parameter_name: str, parameter_value: Any) -> Union[List, List[str]]:
```
[the remark this answers is not in the document]
Comment (williamjamir) [reply hookman/hookman_generator.py]: Done =)

File: tests/test_hookman_generator/generate_plugin.c [hunk 2]
```diff
@@ -1,6 +1,5 @@
 #include "hook_specs.h"
 
-INIT_HOOKS()
 
 // HOOK_FRICTION_FACTOR(v1, v2){}
-// HOOK_FRICTION_FACTOR_2(v1, v2){}
+// HOOK_FRICTION_FACTOR_2(v1, v2){}
```
Comment (oliveira-esss) [inline tests/test_hookman_generator/generate_plugin.c]: "No newline at end of file" is that a problem?

Comment (oliveira-esss) [reply tests/test_hookman_generator/generate_plugin.c]: https://stackoverflow.com/questions/72271/no-newline-at-end-of-file-compiler-warning

Comment (prusse-martin) [reply tests/test_hookman_generator/generate_plugin.c]: https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206

>  **3.206 Line**
> A sequence of zero or more non- <newline> characters plus a terminating <newline> character.

Comment (williamjamir) [reply tests/test_hookman_generator/generate_plugin.c]: Commit 7e57049 fix this issue ;)

**Comment by williamjamir** [comment]
> I feel a bit uneasy with file missing a new line at the end.
> Maybe I am getting old.

Commit 7e57049 fix this issue ;)

**Pull request merged**